system/lib/wasmfs/file.h

```diff
@@ -158,6 +158,13 @@ class Directory : public File {
       lockedInserted.setParent(file);
     }
 
+    void unlinkEntry(std::string pathName) {
+      // Hold lock on target file for duration of the operation.
```

Comment:
This comment says what the code does, but not *why* it does it. I think it's important to stress that it makes the entire operation atomic, that is, the unlinking on both sides happens as if it is a single operation, and no other thread can observe state in the middle.